### PR TITLE
Fixes typeguard to correctly evaluate Error type when globals don't match

### PR DESCRIPTION
**Pull request by scalvert**
The typeguard [added here](https://github.com/stefanpenner/node-fixturify-project/commit/a9ab547e246f57aed6a1a35748650f0b595dc742#diff-dcdc3e0b3362edb8fec2a51d3fa51f8fb8af8f70247e06d9887fa934834c9122R616-R618) seems completely reasonable, but fails in `jest` environments due to https://github.com/facebook/jest/issues/11808. 

I've attempted to reasonable duck type the Error check, but am open to adjustments. Additionally, I'm unsure how to provide a reasonable test for this short of adding a separate test using jest, which would manifest this problem.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -613,8 +613,12 @@ function unwrapPackageName(obj: any, packageName: string): fixturify.DirJSON {
   return getFolder(obj, packageName);
 }
 
+function isError(e: unknown): e is Error {
+  return !!e && typeof e === 'object' && Object.prototype.toString.call(e) === '[object Error]' && 'message' in e;
```
Comment (krisselden) [inline index.ts:617]: it is only null that is typeof 'object', so you can make this `e !== null`

**Review by krisselden — APPROVED** [review]

File: index.ts (hunk 1 above)
Comment (scalvert) [reply index.ts:617]: Ya makes sense.

Comment (stefanpenner) [inline index.ts:617]: Let's simplify this further, `isErrnoException` doesn't really provide much value, we basically just want to ensure two things in this guard:

1. e is an object
2. e is not null

As the conditional clauses this guards manually check `e.code` for specific error codes

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: index.ts (hunk 1 above)
Comment (scalvert) [reply index.ts]: Discussed offline, but we decided to widen the check to only what we care about, in this case that it's an object with a `code` property.

Comment (scalvert) [reply index.ts]: Already done. See latest pushes!

**Pull request merged**